### PR TITLE
feat: [WT-2153] WalletConnect support in Checkout Widgets

### PR DESCRIPTION
**Pull request by dreamoftrees**
WalletConnect integration in Connect Widget and environment configuration.
https://immutable.atlassian.net/browse/WT-2153

# Summary
Preparation for WalletConnect support for mobile wallets in the Checkout Widgets.

### CODE REVIEW

File: packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      });
+
+      ethereumProvider?.once('connect', (data) => {
+        console.log(data);
+        walletConnectModal?.closeModal();
+
+        if (connectCallback) {
+          connectCallback(ethereumProvider);
+        }
+      });
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts:72]: Can we test if this event is fired when connecting MetaMask mobile twice (but not through the restore session method, just two separate connections)?

File: packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+      }
+
+      // Hook into next available display_uri
+      ethereumProvider?.once('display_uri', (data) => {
+        // eslint-disable-next-line no-console
+        console.log('** hook wc display_uri', data);
+        // const pairingTopic = data.split('@')[0].replace('wc:', '');
+        // // eslint-disable-next-line no-console
+        // console.log('** hook pairingTopic', pairingTopic);
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts:48]: We could uncomment this to log out the pairing topic again if needed

File: packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+export interface UseWalletConnectParams {
+  connectCallback: (ethereumProvider: EthereumProvider) => void,
+}
+
+export const useWalletConnect = ({ connectCallback }: UseWalletConnectParams) => {
+  const [walletConnectBusy, setWalletConnectBusy] = useState<boolean>(false);
+  const [ethereumProvider, setEthereumProvider] = useState<EthereumProvider | null>(null);
+  const [walletConnectModal, setWalletConnectModal] = useState<WalletConnectModal | null>(null);
+  // const [displayUri, setDisplayUri] = useState<string | null>(null);
```
Comment (ZacharyCouchman) [inline packages/checkout/widgets-lib/src/lib/hooks/useWalletConnect.ts:14]: Let's remove this if we don't need to store it anymore

**Review by sharonsheah — APPROVED** [review]

**Review by ZacharyCouchman — APPROVED** [review]

**Review by ZacharyCouchman — APPROVED** [review]

**Review by sharonsheah — APPROVED** [review]

File: packages/checkout/widgets-lib/src/widgets/connect/components/WalletList.tsx [hunk 4]
```diff
@@ -1,47 +1,89 @@
...
+    connectDispatch({
+      payload: {
+        type: ConnectActions.SET_PROVIDER,
+        provider: web3Provider,
+      },
+    });
+    connectDispatch({
+      payload: {
+        type: ConnectActions.SET_WALLET_PROVIDER_NAME,
+        walletProviderName: WalletProviderName.METAMASK,
```
Comment (keithbro-imx) [inline packages/checkout/widgets-lib/src/widgets/connect/components/WalletList.tsx:57]: Might be Passport!

**Pull request merged**